packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -392,32 +401,31 @@ function createStripDocumentClosingTagsTransform(): TransformStream<
   Uint8Array,
   Uint8Array
 > {
-  const decoder = new TextDecoder()
   return new TransformStream({
     transform(chunk, controller) {
       // We rely on the assumption that chunks will never break across a code unit.
       // This is reasonable because we currently concat all of React's output from a single
       // flush into one chunk before streaming it forward which means the chunk will represent
       // a single coherent utf-8 string. This is not safe to use if we change our streaming to no
       // longer do this large buffered chunk
-      let originalContent = decoder.decode(chunk)
-      let content = originalContent
-
       if (
-        content === '</body></html>' ||
-        content === '</body>' ||
-        content === '</html>'
+        isEquivalentUint8Arrays(chunk, ENCODED_TAGS.CLOSED.BODY_AND_HTML) ||
+        isEquivalentUint8Arrays(chunk, ENCODED_TAGS.CLOSED.BODY) ||
+        isEquivalentUint8Arrays(chunk, ENCODED_TAGS.CLOSED.HTML)
       ) {
         // the entire chunk is the closing tags.
         return
       } else {
         // We assume these tags will go at together at the end of the document and that
         // they won't appear anywhere else in the document. This is not really a safe assumption
         // but until we revamp our streaming infra this is a performant way to string the tags
-        content = content.replace('</body>', '').replace('</html>', '')
-        if (content.length !== originalContent.length) {
-          return controller.enqueue(encoder.encode(content))
-        }
+        let transformed = removeFromUint8Array(chunk, ENCODED_TAGS.CLOSED.BODY)
+        transformed = removeFromUint8Array(
+          transformed,
+          ENCODED_TAGS.CLOSED.HTML
+        )
+
+        controller.enqueue(transformed)
       }
 
       controller.enqueue(chunk)
```

Comment:
This needs to be deleted otherwise we double-enqueue

Reply:
Ah good catch. I dropped a return statement here. Fixing now and trying the test that is failing

Reply:
It's a nit but can we bring the enqueue inside the else block or drop the else block. The current construction makes it seem like the stripping functions are only sometimes ran before the enqueue when that isn't the case

Reply:
sure